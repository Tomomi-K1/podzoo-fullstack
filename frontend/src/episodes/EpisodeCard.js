import React, { useState, useEffect } from "react";
import removeTags from "../common/helper";
import Typography from "@mui/material/Typography";
import Card from "@mui/material/Card";


/**use as a card in a episode list */
function EpisodeCard({episode}){
    return (
        <Card className={`${episode.id}`}>
                <img src={episode.image} alt="podcast artwork" height={100} width={100}/>
                <Typography variant='body2'>{episode.title}</Typography>
                <Typography variant='body2'>{removeTags(episode.description)}</Typography>
                <figure>
                <figcaption>Published: {episode.datePublishedPretty}</figcaption>
                {/* if not audio, how  can I show no audio? */}
                    <audio
                        controls
                        src={episode.enclosureUrl}>

                </audio>
            </figure>
     
        </Card>
    );
}

export default EpisodeCard;