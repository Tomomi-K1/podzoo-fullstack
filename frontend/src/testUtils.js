import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };