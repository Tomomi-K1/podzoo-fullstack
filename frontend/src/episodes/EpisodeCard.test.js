import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import EpisodeCard from "./EpisodeCard";
import { UserProvider } from "../testUtils";

const episode ={
    id: 1,
    image: 'test-image-url',
    title: 'test title',
    datePublishedPretty:'test-date',
    enclosureUrl: 'test-link-to-audio',
    enclosureType: 'audio/mp3',
    description:'test-description'
    }
it('renders', () => {
    render(
        <MemoryRouter>
            <EpisodeCard episode={episode}/>
        </MemoryRouter>
    ) 
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <EpisodeCard episode={episode}/>
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

it('has correctly renders', () => {
    render(
        <MemoryRouter>
            <EpisodeCard episode={episode} />
        </MemoryRouter>
        )
    expect(screen.getByText('test title')).toBeInTheDocument();
})
