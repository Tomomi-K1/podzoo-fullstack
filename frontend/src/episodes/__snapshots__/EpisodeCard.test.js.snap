// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root 1 css-1ubr38b-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-1nqrw45"
    >
      <img
        alt="episode artwork"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-gganah-MuiCardMedia-root"
        src="test-image-url"
      />
    </div>
    <div
      class="MuiBox-root css-1ylu0bo"
    >
      <div
        class="MuiCardContent-root css-1k2e1n7-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1nebxi1-MuiTypography-root"
        >
          test title
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1yeu3gx-MuiTypography-root"
        >
          Published: test-date
        </p>
        <audio
          controls=""
          src="test-link-to-audio"
          type="audio/mp3"
        />
      </div>
      <button
        aria-expanded="false"
        aria-label="show more"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-3ogxn6-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
